app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partner/PartnerTypeController.scala

```diff
@@ -95,90 +92,48 @@ class PartnerTypeController @Inject() (
-                      getPartnershipRedirectUrl(appConfig.limitedLiabilityPartnershipJourneyUrl,
-                                                partnerId
-                      )
-                        .map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
-                    case Some(SCOTTISH_PARTNERSHIP) =>
-                      getPartnershipRedirectUrl(appConfig.scottishPartnershipJourneyUrl, partnerId)
-                        .map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
-                    case Some(SCOTTISH_LIMITED_PARTNERSHIP) =>
-                      getPartnershipRedirectUrl(appConfig.scottishLimitedPartnershipJourneyUrl,
-                                                partnerId
-                      )
-                        .map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
-                    case _ =>
-                      //TODO later CHARITABLE_INCORPORATED_ORGANISATION & OVERSEAS_COMPANY_NO_UK_BRANCH will have their own not supported page
-                      Future(
-                        Redirect(
-                          organisationRoutes.OrganisationTypeNotSupportedController.onPageLoad()
+    (authenticate andThen journeyAction).async {
+      implicit request =>
+        PartnerType.form()
+          .bindFromRequest()
+          .fold(
+            (formWithErrors: Form[PartnerType]) =>
+              Future(BadRequest(page(formWithErrors, partnerId))),
+            (partnershipPartnerType: PartnerType) =>
+              updatePartnerType(partnershipPartnerType, partnerId).flatMap { _ =>
+                FormAction.bindFromRequest match {
+                  case SaveAndContinue =>
+                    partnershipPartnerType.answer match {
+                      case Some(SOLE_TRADER) =>
+                        getSoleTraderRedirectUrl(appConfig.soleTraderJourneyUrl,
+                                                 appConfig.partnerGrsCallbackUrl(partnerId)
+                        )
+                          .map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
+                      case Some(UK_COMPANY) | Some(OVERSEAS_COMPANY_UK_BRANCH) =>
+                        getUkCompanyRedirectUrl(appConfig.incorpLimitedCompanyJourneyUrl,
+                                                appConfig.partnerGrsCallbackUrl(partnerId)
+                        )
+                          .map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
+                      case Some(LIMITED_LIABILITY_PARTNERSHIP) =>
+                        getPartnershipRedirectUrl(appConfig.limitedLiabilityPartnershipJourneyUrl,
+                                                  appConfig.partnerGrsCallbackUrl(partnerId)
+                        ).map(journeyStartUrl => SeeOther(journeyStartUrl).addingToSession())
+                      case Some(SCOTTISH_PARTNERSHIP) | Some(SCOTTISH_LIMITED_PARTNERSHIP) =>
```

Comment:
I thought we got partnership names for the incorporated partnerships types, like Scottish Limited? We do when I do a Limited Partnership in Staging (for some reason Scottish Limited is causing an error in Staging).